map_test.go

```diff
@@ -99,6 +105,15 @@ func TestIndexMap(t *testing.T) {
 		return true
 	})
 	assert.Equal(t, imap.Len(), count)
+
+	// RangeBy
+	count = 0
+	imap.RangeBy(CityIndex, func(key any, vals []*Person) bool {
+		count++
+		assert.Equal(t, imap.indexes[CityIndex].inner[key].Collect(), vals)
```

Comment:
Equals leads to:
expected: []*indexmap.Person{(*indexmap.Person)(0xc0000749b0), (*indexmap.Person)(0xc000074a00)}
actual  : []*indexmap.Person{(*indexmap.Person)(0xc000074a00), (*indexmap.Person)(0xc0000749b0)}
```suggestion
		assert.EqualValues(t, imap.indexes[CityIndex].inner[key].Collect(), vals)
```
should work IMHO

Reply:
not sure this would work. EqualValues is supposed to compare values of different types e.g. `uint(42)` and `int32(42)`. Anyways, I've added a sort before comparison.